### ISSUE TITLE
A few small issues

### ISSUE THREAD

**Issue by lordnynex** [jekyll/docker#77]
- Building the images is tough because the Gemfile pulls docker-template from github. Somehow this seems to miss (I think) the docker-api dependency. Removing the `{ :github }` bit seems to resolve the issue. Took me a few minutes to figure it out as ruby is not my primary language. I don't know much about ruby packaging. @envygeeks is there perhaps something missing in the repo?

```
jekyll-docker/vendor/cache/docker-template-648a4d5644fe/lib/docker/template/interface.rb:31:in `parse!': uninitialized constant Docker::Template::Interface::OptParse (NameError)
```

- Unfortunately the CI link to the wiki is broken because the wiki seems to be disabled. I'd really like to understand best practices for CI using these images. 

Unrelated, please be aware I intend to fork this repo. Simply put, this build method is fantastic and I'd like it to become the new defacto standard for some of my images. I do, however, have slightly different system dependencies related to nginx. I intend to champion this repo as an excellent example of how to maintain and release images. I would really like to understand the CI bits though. I'm accustomed to using BATS but I could also figure out serverspec. Any hints or pointers?

**Comment by envygeeks** [comment]
We are in a weird spot with both repos right now as `docker-template` was renamed on Friday to `docker-repo` as I was debating splitting it up into 3 pieces since on Friday I finally stabilized the hooks to what I wanted them to be.  I had considered making them each entirely decoupled... not just decoupled locally.  However, I regretted that decision and didn't revert it before you had started playing.

On Monday it's actually being reverted back to `docker-template` (probably) before... sometimes I take the weekends off... sometimes I just itch to program so I skip taking a weekend.  I ultimately decided that at this time there is no reason for them to split them up, I can still use hooks internally for some pieces and when the time comes the code is decoupled enough to split apart quickly too. I'll go ahead and rename the repo right now so everything goes back to how you need it to be (as none of the splitting was done publicly, just the hinting by renaming.

---

On the other stuff.  I am happy that you found it useful! At one point recently I did discuss (quite publicly) using serverspec, however, I found that serverspec had too many quirks and did not have a bug tracker and took the stance that you should fix the bugs in their software [and I do try and help, but that stance is not a good one to me.]  That left a bad taste in my mouth.  We actually use BATS on all my Docker images too, especially on the base images that power this image, we had used BATS before trying serverspec and went back to BATS after serverspec.

On a side note to your note: `docker-template` plans on adding integration for BATS testing where it will boot and image, load the `tests/` folder with `docker copy` (through the API of course) onto the image, install bats and any depends you ask it to and run the tests for you.  That just hasn't made it yet.

**Comment by lordnynex** [comment]
@envygeeks Thanks for the reply. 

Just as you sent this I stumped upon another issue. My Dockerfile now compiles a bit of stuff at build. I seem to be stumbling on the default timeout for Excon (via docker-api). I found swipely/docker-api#106 which seems to have a fix. Is there any way I can get this config down the stack without having to fork/patch docker-(repo|template)?

RE: Your comments. Painful as it may be, maybe a bit of documentation :). It's an excellent tool and I just happened to stumble upon it from digging around this repo's README. I totally get it though as I'm skipping my weekend to contribute elsewhere :). 

Serverspec seems interesting but is a bit more verbose than I prefer for writing tests. BATS is a lot more concise. I'm not sure how it tests would be executed within the image though. Perhaps a build target like 'builder' that includes bash? I've been doing it the phusion-baseimage way and processing an '--enable-ssh' flag on the ubuntu images init. https://github.com/phatlab/docker-or-site-base/blob/develop/test/runner.sh ... Unfortunately though, I find myself here because the ubuntu images are simply too large to be portable.

**Comment by lordnynex** [comment]
For reference I really seem to be hitting this issue during gcc installation. If I re-run several times sometimes it makes it through (I'm assuming different mirrors are handing me the apk at different speeds).

```
(13/67) Installing gcc (5.3.0-r0)
~/.rvm/gems/ruby-2.1.1/gems/docker-api-1.24.1/lib/docker/connection.rb:52:in `rescue in request': read timeout reached (Docker::Error::TimeoutError)
```

**Comment by envygeeks** [comment]
You can send a pull request for this file: https://github.com/envygeeks/docker-template/blob/master/lib/docker/template.rb#L16 and I'll happily accept the write_timeout and increase of read_timeout.  I'll add a configuration option for it on Monday if you file a ticket.

On the documentation part, technically `docker-template` isn't a public project yet, this repo served as it's beta with my repos, it's coming, but not until I fully stabalize the private and public API of `docker-template` this week.  Once that's done a full documentation site will be done explaining everything including why `docker-template` came about.

**Comment by envygeeks** [comment]
On the testing part, we actually plan to enforce Docker 1.7+ (as we do on `docker-template`) so that we can utilize `docker-exec` with tags and labels to get stuff done without having to worry about SSH.  On the bash part, we do plan on adding in bash back to envygeeks/alpine and envygeeks/ubuntu, it's just not ready yet.
